sdk/spring/spring-cloud-azure-actuator-autoconfigure/src/main/java/com/azure/spring/cloud/actuate/autoconfigure/cosmos/CosmosHealthConfiguration.java

```diff
@@ -24,12 +24,12 @@
 @ConditionalOnClass({ CosmosAsyncClient.class, HealthIndicator.class})
 @ConditionalOnBean(CosmosAsyncClient.class)
 @AutoConfigureAfter(AzureCosmosAutoConfiguration.class)
-@ConditionalOnExpression("${spring.cloud.azure.cosmos.enabled:true}")
-@ConditionalOnProperty(prefix = "spring.cloud.azure.cosmos", name = { "endpoint", "database" })
+@ConditionalOnEnabledHealthIndicator("azure-cosmos")
+@ConditionalOnProperty(value = "spring.cloud.azure.cosmos.enabled", havingValue = "true", matchIfMissing = true)
+@ConditionalOnAnyProperty(prefix = "spring.cloud.azure.cosmos", name = "endpoint")
```

Comment:
It should also conditional on the property database?

Reply:
It depends on `@AutoConfigureAfter(AzureCosmosAutoConfiguration.class)`, so it should be the same condition.

Reply:
No, it's not the same, it's possible the AzureCosmosAutoConfiguration will not be enabled.

Reply:
If we only need a @ConditionalOnBean(CosmosAsyncClient.class), we can skip these conditions: 
```java
@ConditionalOnProperty(value = "spring.cloud.azure.cosmos.enabled", havingValue = "true", matchIfMissing = true)
@ConditionalOnAnyProperty(prefix = "spring.cloud.azure.cosmos", name = "endpoint")
```